### PR TITLE
Login event trigger and logout reason

### PR DESCRIPTION
**Pull request by su-chang**
Related issue: #81

### CODE REVIEW

**Review by bung87 — APPROVED** [review]

File: src/manager.ts [hunk 1]
```diff
@@ -164,6 +163,7 @@ export class Manager extends EventEmitter {
       })
     })
     await Promise.all(all)
+    this.emit('login', whatsapp.info.wid._serialized)
```
Comment (windmemory) [inline src/manager.ts:166]: Is there a way to get only the logged in contact data? If we move this code here, the login will be emitted with big delay.

**Review by windmemory — DISMISSED** [review]

File: src/manager.ts (hunk 1 above)
Comment (su-chang) [reply src/manager.ts:166]: OK

**Review by guo40020 — APPROVED** [review]

**Review by wedreamer — DISMISSED** [review]

File: examples/ding-dong-bot.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ void (async () => {
     .catch(async e => {
       console.error('Bot start() fail:', e)
       await puppet.stop()
-      process.exit(-1)
+      // process.exit(-1)
```
Comment (bung87) [inline examples/ding-dong-bot.ts:85]: is this necessary ?

**Review by hcfw007 — DISMISSED** [review]

File: examples/ding-dong-bot.ts (hunk 2 above)
Comment (hcfw007) [reply examples/ding-dong-bot.ts:85]: puppet should never call process.exit

Comment (su-chang) [reply examples/ding-dong-bot.ts:85]: Both are ok, just example.

**Review by bung87 — APPROVED** [review]

**Review by bung87 — APPROVED** [review]

**Review by wedreamer — APPROVED** [review]

**Review by hcfw007 — APPROVED** [review]

**Pull request merged**